src/plugins/typedoc.ts

```diff
@@ -0,0 +1,118 @@
+ * Licensed under the BSD-3 License as modified (the “License”); you may obtain
+ * a copy of the license in the LICENSE and PATENTS files in the root of this
+ * repository.
+ */
+
+import * as Typedoc from "typedoc";
+import {
+    ITsClassEntry,
+    ITsPropertyEntry,
+    ITypedocPluginData,
+} from "../client";
+import { ICompiler, IFile, IPlugin } from "./plugin";
+
+class TypedocApp extends Typedoc.Application {
+    public fromFiles(files: string[]) {
+        const expanded = this.expandInputFiles(files);
+        const project = this.convert(expanded);
+        return project.toObject();
+    }
+
+    // this tricks typedoc into working
+    get isCLI(): boolean {
+        return true;
+    }
+}
+
+export class TypedocPlugin implements IPlugin<ITypedocPluginData> {
+    public compile(files: IFile[], compiler: ICompiler) {
+        const typedoc = new TypedocApp({logger: "none"}).fromFiles(files.map((f) => f.path));
+
+        const classes: {[key: string]: ITsClassEntry} = {};
+        this.visitKind(typedoc, "Class", (classDef: any) => {
+            const name: string = classDef.name;
+
+            let documentation = "";
+            if (classDef && classDef.comment && classDef.comment.shortText) {
+                documentation = classDef.comment.shortText;
+            }
+
+            const classEntry: ITsClassEntry = {
+                documentation: this.renderComment(classDef, compiler),
+                methods: [],
+                name,
+                properties: [],
+                tags: {},
+                type: "class",
+            };
+
+            this.visitKind(classDef, "Property", (def: any) => {
+                const entry: ITsPropertyEntry = {
+                    documentation: this.renderComment(def, compiler),
+                    name: def.name,
+                    tags: this.getTags(def),
+                    // TODO this doesnt work for some types types. we need a
+                    // conversion here to good formattable object
+                    type: def.type.name,
+                };
+                classEntry.properties.push(entry);
+            });
+
+            this.visitKind(classDef, "Method", (def: any) => {
+                const sig = (def && def.signatures && def.signatures[0]) ? def.signatures[0] : def;
+                const entry: ITsPropertyEntry = {
+                    documentation: this.renderComment(sig, compiler),
+                    name: def.name,
+                    tags: this.getTags(sig),
+                    // TODO add type parameters here
+                    type: "method",
+                };
+                classEntry.properties.push(entry);
+            });
+
+            classes[name] = classEntry;
+        });
+
+        return { typedoc: classes };
+    }
+
+    private getTags(obj: any) {
+        return (obj && obj.tags) ? obj.tags : {};
+    }
+
+    /**
+     * Converts a typedoc comment object to a rendered `IBlock`.
```

Comment:
looks like typedoc pull out all tags, but fortunately the content of them is maintained so we can rebuild the string and pass it to render block